.circleci/config.yml

```diff
@@ -284,29 +292,56 @@ jobs:
           environment:
             OMP_NUM_THREADS: 1
 
-      - run: &examples-dask-ml
-          name: examples-dask-ml
-          command: |
-            . venv/bin/activate
-            for file in `find examples -name 'dask_ml*.py'`
-            do
-               python $file
-            done
-
   examples-python36:
     docker:
       - image: circleci/python:3.6
     steps:
-      [checkout, run: *install, run: *install-examples, run: *examples, run: *examples-pytorch-lightning, run: *examples-mn, run: *examples-dask-ml]
+      [checkout, run: *install, run: *install-examples, run: *examples, run: *examples-mn]
 
   examples-python35:
     docker:
       - image: circleci/python:3.5
     steps:
-      [checkout, run: *install, run: *install-examples, run: *examples, run: *examples-mn, run: *examples-dask-ml]
+      - checkout
+
+      - run: *install
+
+      - run: *install-examples
+
+      - run:
+          <<: *examples
+          command: |
+            . venv/bin/activate
+            for file in `find examples -name '*.py' -not -name 'chainermn_*.py' -not -name 'pytorch_lightning_simple.py'`
+            do
+               python $file
+               if grep -e '\-\-pruning' $file > /dev/null; then
+                  python $file --pruning
+               fi
+            done
+
+      - run: *examples-mn
 
   examples-python27:
     docker:
       - image: circleci/python:2.7
     steps:
-      [checkout, run: *install, run: *install-examples, run: *examples, run: *examples-mn]
+      - checkout
+
+      - run: *install
+
+      - run: *install-examples
+
+      - run:
+          <<: *examples
+          command: |
```

Comment:
The new YAML file seems far better than the current one.
But, I think that the examples part can be improved a bit by using `grep` and an environment variable that manages file names to be ignored (see the following diff):
```diff
diff --git a/.circleci/config.yml b/.circleci/config.yml
index 3f40407..9ce1aae 100644
--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -269,7 +269,7 @@ jobs:
           name: examples
           command: |
             . venv/bin/activate
-            for file in `find examples -name '*.py' -not -name 'chainermn_*.py'`
+            for file in `find examples -name '*.py' | grep -vE "$IGNORES"`
             do
                python $file
                if grep -e '\-\-pruning' $file > /dev/null; then
@@ -278,6 +278,7 @@ jobs:
             done
           environment:
             OMP_NUM_THREADS: 1
+            IGNORES: chainermn_.*
 
       - run: &examples-mn
           name: examples-mn
@@ -310,15 +311,9 @@ jobs:
 
       - run:
           <<: *examples
-          command: |
-            . venv/bin/activate
-            for file in `find examples -name '*.py' -not -name 'chainermn_*.py' -not -name 'pytorch_lightning_simple.py'`
-            do
-               python $file
-               if grep -e '\-\-pruning' $file > /dev/null; then
-                  python $file --pruning
-               fi
-            done
+          environment:
+            OMP_NUM_THREADS: 1
+            IGNORES: chainermn_.*|pytorch_lightning.*
 
       - run: *examples-mn
 
@@ -334,14 +329,8 @@ jobs:
 
       - run:
           <<: *examples
-          command: |
-            . venv/bin/activate
-            for file in `find examples -name '*.py' -not -name 'chainermn_*.py' -not -name 'pytorch_lightning_*.py' -not -name 'xgboost_*.py'`
-            do
-               python $file
-               if grep -e '\-\-pruning' $file > /dev/null; then
-                  python $file --pruning
-               fi
-            done
+          environment:
+            OMP_NUM_THREADS: 1
+            IGNORES: chainermn_.*|pytorch_lightning.*|xgboost_.*
 
       - run: *examples-mn
```

Reply:
That is a lot cleaner, I applied your suggestion. I was thinking of prefixing `IGNORES` to avoid collisions but maybe it's okay to tackle it once that becomes relevant.